perl/lib/Sanger/CGP/Star/Implement.pm

```diff
@@ -598,6 +599,23 @@ sub _which {
 	return $path;
 }
 
+sub _to_commandline_safe_tag_for_star {
+	# according to this: https://unix.stackexchange.com/a/398649
+	# preserving tag values is complicated
+	my $tag = shift;
+	# first skip back slashes
+	$tag =~ s/\\/\\\\/g;
+	# then skip other needed-to-skip characters.
+	# this will inevitably introduce an extra back slash into the tag if there's an '!', but for now no other way to skip '!' to prevent bash history expansion.
+	my @need_to_skips = ('$', '`', '"', '!');
```

Comment:
I think this will allow interpolation of variables as you've not escaped `$`:

```bash
$ echo "thing's "are "looking up $funny"
funny: Undefined variable.
```

Reply:
You'd need to test a string with each of those 4 characters in it and confirm it comes out as expected at the other end

Reply:
Ah, I've just realised what the problem is.  The array should be `@need_to_escape`, the name gives the wrong impression of the intent.